services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler.py

```diff
@@ -248,6 +249,34 @@ async def mark_service_for_removal(
             current.dynamic_sidecar.service_removal_state.mark_to_remove(can_save)
             await update_scheduler_data_label(current)
 
+            # cancel current observation task
+            if service_name in self._service_observation_task:
+                service_task: Optional[
+                    Union[asyncio.Task, object]
+                ] = self._service_observation_task[service_name]
+                if isinstance(service_task, asyncio.Task):
+                    service_task.cancel()
+
+                    async def _wait_task(task: asyncio.Task) -> None:
+                        await task
+
+                    with suppress(asyncio.CancelledError):
+                        try:
+                            await asyncio.wait_for(_wait_task(service_task), timeout=10)
+                        except asyncio.TimeoutError:
+                            pass
```

Comment:
Q: what happens if the timeoutError occurs? is there something that will continue running in the background?

Reply:
No the task is already cancelled.